app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileScreen.kt

```diff
@@ -0,0 +1,398 @@
+ * @author Ayman Bakiri
+ * @author Menzo Bouaissi
+ * @since 1.0
+ * @last update 1.0
+ */
+@Composable
+fun ProfileScreen(viewModel: ProfileViewModel) {
+  val currentProfileState by viewModel.profile.collectAsState()
+  val isInEditMode by viewModel.isInEditMode.collectAsState()
+
+  val currentProfile: Profile = currentProfileState
+
+  if (isInEditMode) {
+    EditableVisitCard(
+        profile = currentProfile,
+        onProfileChange = { updatedProfile -> viewModel.updateProfile(updatedProfile) },
+        viewModel = viewModel)
+  } else {
+    Column(modifier = Modifier.fillMaxSize().padding(16.dp).testTag("profileScreen")) {
+      Box(modifier = Modifier.fillMaxWidth()) {
+        VisitCard(Modifier, currentProfile)
+        ProfileSwitch(Modifier.align(Alignment.TopEnd), viewModel)
+        ClickableIcon(Modifier.align(Alignment.BottomEnd), Icons.Filled.Create, viewModel)
+      }
+    }
+  }
+}
+/**
+ * This handle the logic behind the switch that can permit the user to switch to the anonymous mode
+ *
+ * @param modifier to place the switch at a place, and still be able to modify it.
+ * @author Menzo Bouaissi
+ * @since 1.0
+ * @last update 1.0
+ */
+@Composable
+fun ProfileSwitch(modifier: Modifier = Modifier, viewModel: ProfileViewModel) {
+  // Determine the current public mode state
+  val isPublicMode by viewModel.isInPublicMode.collectAsState(false)
+  Switch(
+      checked = isPublicMode,
+      onCheckedChange = {
+        // When the switch is toggled, call viewModel's method to update the profile's public mode
+        viewModel.togglePublicMode()
+      },
+      modifier =
+          modifier
+              .graphicsLayer {
+                scaleX = SCALE_X
+                scaleY = SCALE_Y
+              }
+              .testTag("profileSwitch"),
+      colors =
+          SwitchDefaults.colors(
+              checkedThumbColor = MaterialTheme.colorScheme.primary,
+              checkedTrackColor = MaterialTheme.colorScheme.primaryContainer,
+              uncheckedThumbColor = MaterialTheme.colorScheme.secondary,
+              uncheckedTrackColor = MaterialTheme.colorScheme.secondaryContainer,
+          ),
+  )
+}
+
+/**
+ * Handle the logic behind the edit button
+ *
+ * @param modifier the modifier to be applied to the Icon
+ * @param icon the icon to be display
+ * @param viewModel the viewModel that will handle the profile
+ * @author Menzo Bouaissi
+ * @since 1.0
+ * @last update 1.0
+ */
+@Composable
+fun ClickableIcon(
+    modifier: Modifier,
+    icon: ImageVector,
+    viewModel: ProfileViewModel,
+) {
+
+  IconButton(
+      modifier = modifier.then(Modifier.size(24.dp)).testTag("clickableIcon"),
+      onClick = { viewModel.toggleEditMode() }) {
+        Icon(
+            icon,
+            contentDescription = "Edit",
+            modifier = modifier,
+            // tint = if (isInEditMode) MaterialTheme.colorScheme.onSurface else
```

Comment:
We agreed on getting rid of commented out codes and just put it back later on if we need it since github has the history.